### PR TITLE
postgres

### PR DESCRIPTION
**Pull request by joshuacox**
added a postgres version

### CODE REVIEW

File: templates/drupal/1/rancher-compose.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    - variable: "DB_PASS"
+      description: "Database Password to use for the server"
+      label: "DB Password"
+      required: true
+      default: ""
+      type: "password"
+
+drupal:
+  retain_ip: true
```
Comment (rawmind0) [inline templates/drupal/1/rancher-compose.yml]: Healthcheck??

File: templates/drupal/1/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      POSTGRES_PASSWORD:  ${DB_PASS}
+      POSTGRES_DB:  ${DB_NAME}
+    volumes_from:
+      - drupal-datavolume
+    restart: always
+
+  drupal-datavolume:
+    image: "busybox"
+    volumes:
+      - /var/www/drupal
```
Comment (rawmind0) [inline templates/drupal/1/docker-compose.yml]: This volume definition is wrong. Drupal recommends 4 different volumes, https://hub.docker.com/_/drupal/
```
    volumes:
      - drupal-modules:/var/www/html/modules
      - drupal-profiles:/var/www/html/profiles
      - drupal-themes:/var/www/html/themes
      - drupal-sites:/var/www/html/sites
```

File: templates/drupal/1/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    links:
+      - postgres:postgres
+    volumes_from:
+      - drupal-datavolume
+    restart: always
+
+  postgres:
+    labels:
+      io.rancher.sidekicks: drupal-datavolume
+      io.rancher.container.pull_image: always
```
Comment (rawmind0) [inline templates/drupal/1/docker-compose.yml]: If you use concrete versions, you don't need pull always.

File: templates/drupal/1/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    ports:
+      - ${public_port}:80
+    links:
+      - postgres:postgres
+    volumes_from:
+      - drupal-datavolume
+    restart: always
+
+  postgres:
```
Comment (rawmind0) [inline templates/drupal/1/docker-compose.yml]: Postgres shouldn't be a sidekick. Also, if you use "volumes_from", postgres would need another busybox  just for him.

File: templates/drupal/1/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+version: '2'
+
+services:
+
+  drupal:
+    image: drupal:8.2-apache
+    labels:
+      io.rancher.sidekicks: postgres,drupal-datavolume
+      io.rancher.container.pull_image: always
+    ports:
```
Comment (rawmind0) [inline templates/drupal/1/docker-compose.yml]: Better expose ports using lb.

File: templates/drupal/1/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+version: '2'
+
+services:
+
+  drupal:
+    image: drupal:8.2-apache
+    labels:
+      io.rancher.sidekicks: postgres,drupal-datavolume
+      io.rancher.container.pull_image: always
```
Comment (rawmind0) [inline templates/drupal/1/docker-compose.yml]: If you use concrete versions, you don't need pull always.

**Comment by rawmind0** [comment]
Hi @joshuacox, 

thanks for the update. I've tested your package and it's working, but some errors in defintions, please take a look at my comments.
Also, once drupal with postgres is deloyed, you get the "update_available" message. 

This is not the best approach to provide the options to select mysql or postgres, it's better to use go templating in docker-compose.yml.tpl . 

I've made a version applying best practices and using go templating.  If you like to try, add my repo https://github.com/rawmind0/service-catalog  to your rancher and deploy version 8.3.5-rancher1

Please, take a look the definition...https://github.com/rawmind0/service-catalog/tree/master/templates/drupal/1

Its working well, choosing mysql and postgres.

**Comment by joshuacox** [comment]
@rawmind0 the go templating is very cool, and answers many questions I had in many other areas.  Cheers!

**Comment by rawmind0** [comment]
Thanks @joshuacox... I'm glad it helps you... :)

LGTM...

**Pull request merged**